system/jlib/jmetrics.hpp

```diff
@@ -218,14 +218,20 @@ class jlib_decl PeriodicMetricSink : public MetricSink
 
 extern "C" { typedef hpccMetrics::MetricSink* (*getSinkInstance)(const char *, const IPropertyTree *pSettingsTree); }
 
-struct SinkInfo;
+struct SinkInfo
+{
+    explicit SinkInfo(MetricSink *_pSink) : pSink{_pSink} {}
+    MetricSink *pSink = nullptr;             // ptr to the sink
```

Comment:
should the destructor delete the sink, otherwise it will leak?  Possible destruction issues at program termination if it did.